pytorch_lightning/trainer/connectors/data_connector.py

```diff
@@ -435,7 +435,13 @@ def _reset_eval_dataloader(
             )
 
         # add samplers
-        dataloaders = [self._prepare_dataloader(dl, False, mode=mode) for dl in dataloaders if dl is not None]
+        dataloaders = [
+            self._prepare_dataloader(
+                dl, shuffle=hasattr(dl, "sampler") and isinstance(dl.sampler, RandomSampler), mode=mode
```

Comment:
Shouldn't this use the same conditions as `_check_eval_shuffling`?

Reply:
yes, though so but if its a custom sampler, then it would set shuffle=True, which might be incorrect.

Reply:
ok, looks like we raise an error in the case of the custom sampler already. will update.